### PR TITLE
Cleanup from variable stat size

### PR DESCRIPTION
**Pull request by ggreenway**
- Add an integration test that verifies the the results of --hot-restart-version and
  the corresponding value on the admin endpoint match
- Documentation for CLI flags
- Add additional check to unittest that got lost yesterday

Signed-off-by: Greg Greenway <ggreenway@apple.com>

### CODE REVIEW

File: test/integration/hotrestart_test.sh [hunk 1]
```diff
@@ -65,6 +65,35 @@ do
...
+  CLI_HOT_RESTART_VERSION=`${ENVOY_BIN} --hot-restart-version 2>&1`
+  if [ "${ADMIN_HOT_RESTART_VERSION}" != "${CLI_HOT_RESTART_VERSION}" ]; then
+      echo "Hot restart version mismatch: ${ADMIN_HOT_RESTART_VERSION} != ${CLI_HOT_RESTART_VERSION}"
+      exit 2
+  fi
+
+  echo "Checking max-stat-name-len"
+  CLI_HOT_RESTART_VERSION=`${ENVOY_BIN} --hot-restart-version --max-stat-name-len 1234 2>&1`
+  if [ "${ADMIN_HOT_RESTART_VERSION}" = "${CLI_HOT_RESTART_VERSION}" ]; then
+      echo "Hot restart version match when it should mismatch: ${ADMIN_HOT_RESTART_VERSION} != ${CLI_HOT_RESTART_VERSION}"
```
Comment (mattklein123) [inline test/integration/hotrestart_test.sh]: nit: please break this line and the one below at 100 cols

File: docs/operations/cli.rst [hunk 2]
```diff
@@ -116,3 +116,12 @@ following are the command line options that Envoy supports.
...
   during a hot restart. See the :ref:`hot restart overview <arch_overview_hot_restart>` for more
   information. Defaults to 900 seconds (15 minutes).
+
+.. option:: --max-stat-name-len <uint64_t>
+
+  *(optional)* The maximum name length (in bytes) for a stat.  Instances of Envoy must have the
+  same value for this option in order to hot-restart. Defaults to 127.
+
+.. option:: --max-stats <uint64_t>
+  *(optional)* The maximum number of stats that can be shared between hot-restarts. Instances of
```
Comment (mattklein123) [inline docs/operations/cli.rst]: nit: line break before this line to match the others.

File: docs/operations/cli.rst [hunk 3]
```diff
@@ -116,3 +116,12 @@ following are the command line options that Envoy supports.
   *(optional)* The time in seconds that Envoy will wait before shutting down the parent process
   during a hot restart. See the :ref:`hot restart overview <arch_overview_hot_restart>` for more
   information. Defaults to 900 seconds (15 minutes).
+
+.. option:: --max-stat-name-len <uint64_t>
+
+  *(optional)* The maximum name length (in bytes) for a stat.  Instances of Envoy must have the
+  same value for this option in order to hot-restart. Defaults to 127.
```
Comment (htuch) [inline docs/operations/cli.rst]: Nit: is it "hot restart" or "hot-restart" for consistency with rest of docs?

File: docs/operations/cli.rst [hunk 4]
```diff
@@ -116,3 +116,12 @@ following are the command line options that Envoy supports.
...
+
+.. option:: --max-stats <uint64_t>
+  *(optional)* The maximum number of stats that can be shared between hot-restarts. Instances of
+  Envoy must have the same value for this option in order to hot-restart. Defaults to 16384.
```
Comment (htuch) [inline docs/operations/cli.rst]: Can you elaborate below on the UX if there's a mismatch? I.e. version mismatch.

File: test/integration/hotrestart_test.sh [hunk 5]
```diff
@@ -65,6 +65,35 @@ do
   kill -SIGHUP ${FIRST_SERVER_PID}
   sleep 3
 
+  # The heapchecker outputs some data to stderr on every execution.  This gets intermingled
+  # with the output from --hot-restart-version, so disable the heap-checker for these runs.
+  SAVED_HEAPCHECK=${HEAPCHECK}
+  unset HEAPCHECK
+
+  ADMIN_ADDRESS_0=`cat ${ADMIN_ADDRESS_PATH_0}`
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: Nit: Prefer `$(command stuff)` style in Envoy scripts.

File: test/integration/hotrestart_test.sh [hunk 6]
```diff
@@ -65,6 +65,35 @@ do
...
+  ADMIN_HOT_RESTART_VERSION=`curl http://${ADMIN_ADDRESS_0}/hot_restart_version 2>/dev/null`
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: Maybe use `-s`?

File: test/integration/hotrestart_test.sh [hunk 7]
```diff
@@ -65,6 +65,35 @@ do
...
+  CLI_HOT_RESTART_VERSION=`${ENVOY_BIN} --hot-restart-version 2>&1`
+  if [ "${ADMIN_HOT_RESTART_VERSION}" != "${CLI_HOT_RESTART_VERSION}" ]; then
+      echo "Hot restart version mismatch: ${ADMIN_HOT_RESTART_VERSION} != ${CLI_HOT_RESTART_VERSION}"
+      exit 2
+  fi
+
+  echo "Checking max-stat-name-len"
+  CLI_HOT_RESTART_VERSION=`${ENVOY_BIN} --hot-restart-version --max-stat-name-len 1234 2>&1`
+  if [ "${ADMIN_HOT_RESTART_VERSION}" = "${CLI_HOT_RESTART_VERSION}" ]; then
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: Nit: Style guide prefers `[[`, see https://google.github.io/styleguide/shell.xml?showone=Test,_%5B_and_%5B%5B#Test,_%5B_and_%5B%5B.

File: test/integration/hotrestart_test.sh [hunk 8]
```diff
@@ -70,25 +70,29 @@ do
   SAVED_HEAPCHECK=${HEAPCHECK}
   unset HEAPCHECK
 
-  ADMIN_ADDRESS_0=`cat ${ADMIN_ADDRESS_PATH_0}`
-  ADMIN_HOT_RESTART_VERSION=`curl http://${ADMIN_ADDRESS_0}/hot_restart_version 2>/dev/null`
-  CLI_HOT_RESTART_VERSION=`${ENVOY_BIN} --hot-restart-version 2>&1`
-  if [ "${ADMIN_HOT_RESTART_VERSION}" != "${CLI_HOT_RESTART_VERSION}" ]; then
-      echo "Hot restart version mismatch: ${ADMIN_HOT_RESTART_VERSION} != ${CLI_HOT_RESTART_VERSION}"
+  echo "Checking for match of --hot-restart-version and admin /hot_restart_version"
+  ADMIN_ADDRESS_0=$(cat ${ADMIN_ADDRESS_PATH_0})
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: `cat "${ADMIN_ADDRESS_PATH}"`, thanks.

File: test/integration/hotrestart_test.sh [hunk 9]
```diff
@@ -70,25 +70,29 @@ do
...
+  ADMIN_HOT_RESTART_VERSION=$(curl -s http://${ADMIN_ADDRESS_0}/hot_restart_version)
+  CLI_HOT_RESTART_VERSION=$(${ENVOY_BIN} --hot-restart-version 2>&1)
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: `$("${ENVOY_BIN" ...`

File: test/integration/hotrestart_test.sh [hunk 10]
```diff
@@ -70,25 +70,29 @@ do
...
+      echo "Hot restart version mismatch: ${ADMIN_HOT_RESTART_VERSION} != " \
+           "${CLI_HOT_RESTART_VERSION}"
       exit 2
   fi
 
   echo "Checking max-stat-name-len"
-  CLI_HOT_RESTART_VERSION=`${ENVOY_BIN} --hot-restart-version --max-stat-name-len 1234 2>&1`
-  if [ "${ADMIN_HOT_RESTART_VERSION}" = "${CLI_HOT_RESTART_VERSION}" ]; then
-      echo "Hot restart version match when it should mismatch: ${ADMIN_HOT_RESTART_VERSION} != ${CLI_HOT_RESTART_VERSION}"
+  CLI_HOT_RESTART_VERSION=$(${ENVOY_BIN} --hot-restart-version --max-stat-name-len 1234 2>&1)
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: Ditto

**Pull request merged**